src/Microsoft.Identity.Client/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -65,6 +65,7 @@ internal MsalAccessTokenCacheItem
             CachedAt = CoreHelpers.CurrDateTimeInUnixTimestamp();
             RawClientInfo = rawClientInfo;
 
+            HomeAccountId = userId;
```

Comment:
>HomeAccountId = userId; [](start = 12, length = 23)

I'm not sure this is a good idea

Reply:
Can you pls. explain why you are doing this? It may be relevant to make a comment on why this is needed for ADFS

---
In reply to: [252132011](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/834#discussion_r252132011) [](ancestors = 252132011)

Reply:
Adfs doesnt send back client info, so this value needs to be explicitly set. I can add a comment

Reply:
I can't see how this is not potentially breaking non ADFS scenarios. Is it explicitly set in all other cases?

I suppose this is one of the scenarios where I would prefer a factory mehtod to set the HomeAccountId to null when-ever this is an ADFS scenario

---
In reply to: [255629863](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/834#discussion_r255629863) [](ancestors = 255629863)